gateway/src/main/java/io/zeebe/gateway/grpc/GrpcErrorMapper.java

```diff
@@ -113,22 +105,20 @@ private Status mapBrokerErrorToStatus(final BrokerError error) {
         break;
       case RESOURCE_EXHAUSTED:
         builder.setCode(Code.RESOURCE_EXHAUSTED_VALUE);
+        logger.trace("Target broker is currently overloaded: {}", error);
         break;
       case PARTITION_LEADER_MISMATCH:
         // return UNAVAILABLE to indicate to the user that retrying might solve the issue, as this
         // is usually a transient issue
+        logger.trace("Target broker was not the leader of the partition: {}", error);
         builder.setCode(Code.UNAVAILABLE_VALUE);
         break;
-        // all the following are not errors which retrying (with the same gateway) will solve
-      case INVALID_MESSAGE_TEMPLATE:
```

Comment:
Normally I'd prefer having all cases, but in this case the non default cases are more the exceptions where we want different behaviour, so I think it makes sense to remove all explicit cases.